libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -346,39 +348,46 @@ void PbftManager::resetStep_() {
 }
 
 bool PbftManager::resetRound_() {
-  // Jump to round
-  auto consensus_pbft_round = vote_mgr_->roundDeterminedFromVotes(TWO_T_PLUS_ONE);
+  auto determined_round_and_period = vote_mgr_->determineRoundAndPeriodFromVotes(TWO_T_PLUS_ONE);
+  if (!determined_round_and_period.has_value()) {
+    return false;
+  }
+
+  auto [determined_round, determined_previous_round_period] = determined_round_and_period.value();
```

Comment:
if you do check for optional value, do not use `.value()` but * as it has perf improvements